File: Medium/2026-05-17-1306-Jump-Game-III/solution.py
class Solution:
    def canReach(self, arr: List[int], start: int) -> bool:
        n = len(arr)
        visited = set()
        stack = [start]
        
        while stack:
            index = stack.pop()
            if index < 0 or index >= n or index in visited:
                continue
            visited.add(index)
            if arr[index] == 0:
                return True
            stack.append(index + arr[index])
            stack.append(index - arr[index])
        
        return False

File: Medium/2026-05-17-1306-Jump-Game-III/solution.js
var canReach = function(arr, start) {
    const n = arr.length;
    const visited = new Set();
    
    const dfs = (index) => {
        if (index < 0 || index >= n || visited.has(index)) return false;
        if (arr[index] === 0) return true;
        
        visited.add(index);
        
        return dfs(index + arr[index]) || dfs(index - arr[index]);
    };
    
    return dfs(start);
};